include/DoocsBackendNumericRegisterAccessor.h

```diff
@@ -0,0 +1,371 @@
+      case DATA_A_BOOL:
+      case DATA_A_SHORT: {
+        copyFromSourcePointer(this->dst.get_short_array());
+        return;
+      }
+      case DATA_IFFF: {
+        for(size_t i = 0; i < this->nElements; i++) {
+          if(i + this->elementOffset == 0) {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<int>(i + this->elementOffset));
+          }
+          else {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<float>(i + this->elementOffset));
+          }
+        }
+        return;
+      }
+      default:
+        // inefficient copying via single element access for other data types (including scalars)
+        callForDoocsType(this->dst, [&](auto t) {
+          using T = decltype(t);
+          for(size_t i = 0; i < this->nElements; i++) {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<T>(i + this->elementOffset));
+          }
+        });
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  template<>
+  inline void DoocsBackendNumericRegisterAccessor<ChimeraTK::Void>::doPreWrite(
+      TransferType type, VersionNumber version) {
+    DoocsBackendRegisterAccessor<ChimeraTK::Void>::doPreWrite(type, version);
+
+    src.set(0);
+  }
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  void DoocsBackendNumericRegisterAccessor<UserType>::doPreWrite(TransferType type, VersionNumber version) {
+    DoocsBackendRegisterAccessor<UserType>::doPreWrite(type, version);
+
+    // implement read-modify-write
+    if(this->isPartial) {
+      this->doReadTransferSynchronously();
+      this->src = this->dst;
+      // make sure the array is long enough, as the remote server might have changed the length on its side
+      if(size_t(this->src.length()) < this->nElements + this->elementOffset) {
+        this->src.length(this->nElements + this->elementOffset);
+      }
+    }
+
+    // define lambda used below for optimisation
+    auto copyToTargetPointer = [&](auto* targetPointer) {
+      using TargetType = std::remove_reference_t<decltype(*targetPointer)>;
+      assert(targetPointer);
+
+      targetPointer += this->elementOffset;
+      if constexpr(std::is_same<UserType, TargetType>::value) {
+        // raw and target data type match, do a memcopy
+        memcpy(targetPointer, this->buffer_2D[0].data(), this->nElements * sizeof(TargetType));
+      }
+      else {
+        std::transform(this->buffer_2D[0].begin(), this->buffer_2D[0].end(), targetPointer,
+            [](UserType v) { return ChimeraTK::userTypeToNumeric<TargetType>(v); });
+      }
+    };
+
+    // optimise depending on type
+    switch(this->src.type()) {
+      case DATA_SPECTRUM:
+      case DATA_A_FLOAT: {
+        copyToTargetPointer(this->src.get_float_array());
```

Comment:
Can you set data using the ```get_float_array()``` function?

Reply:
Why not? It returns the pointer to the underlying array.